packages/kbn-alerts-ui-shared/src/common/hooks/use_fetch_alerts_fields_query.ts

```diff
@@ -37,10 +37,12 @@ export const useFetchAlertsFieldsQuery = (
   );
 
   return useQuery({
-    queryKey: queryKeyPrefix.concat(JSON.stringify(featureIds)),
+    queryKey: queryKeyPrefix.concat(JSON.stringify(featureIds.sort())),
```

Comment:
Not sure if it's best to always sort while building the query key or avoiding sort in other places. I noticed a duplicated request where only the order of the feature ids changed because of [this sort](https://github.com/elastic/kibana/pull/189638/files#diff-8e7de7345f3f13823c76709ff7d5a53f65664d2a12dcea8fa686291282457b4fL30)

Reply:
The reason this happened is because we stringify the feature IDs array. There is no need to do that. We could pass the array to the query fix directly as `['alerts', fetchAlertsFields.name, featureIds]`. React query is smart enough to handle the [reordering](https://tanstack.com/query/v4/docs/framework/react/guides/query-keys#query-keys-are-hashed-deterministically).

Reply:
Ah great! Do you think we [should remove this](https://github.com/elastic/kibana/blob/305d7f4586fb7c62c5891a3db43fdbf6c76b9ad2/x-pack/plugins/triggers_actions_ui/public/application/hooks/use_alert_data_view.ts#L30) regardless of query keys? It seems more problematic than useful to me

Reply:
Yes, I do not think it provides any benefit to the code.